flutter/example/lib/main.dart

```diff
@@ -59,6 +59,11 @@ class MainScaffold extends StatelessWidget {
         child: Column(
           children: [
             const Center(child: Text('Trigger an action:\n')),
+            RaisedButton(
```

Comment:
there are other buttons using the `handleError` method, are they still needed or caught automatically?
if automatically, we should remove it and add a comment, that if appRunner is there, we get them automatically, otherwise `catchError` would be necessary.

Reply:
type 'Future<void>' is not a subtype of type 'Future<Null>?'

Tested with Flutter `1.25.0-8.3.pre` and `Flutter 1.22.6`

For all thrown errors, the `catchError` method is called.

- `Dart: async throws`
- `Dart: Fail in microtask`
- `Dart: Fail in compute`

For the last two, there is also an error reported in the runZonedGuarded onError:

`type 'Future<void>' is not a subtype of type 'Future<Null>?'`

For the new sample, I have added a comment explaining where throwing futures are handled when not implementing `catchError`.

Reply:
cool, just remove the `catchError` call if `runZonedGuarded` catches it automatically.

> type 'Future<void>' is not a subtype of type 'Future<Null>?'

not sure if I follow, does it mean that `runZonedGuarded` catches it but the exception is wrong?